src/Relations/MorphToMany.php

```
@@ -0,0 +1,461 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Laravel\Relations;
+
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\MorphToMany as EloquentMorphToMany;
+use Illuminate\Support\Arr;
+
+use function array_diff;
+use function array_key_exists;
+use function array_keys;
+use function array_map;
+use function array_merge;
+use function array_reduce;
+use function array_values;
+use function count;
+use function is_array;
+use function is_numeric;
+use function is_string;
+use function str_contains;
+
+class MorphToMany extends EloquentMorphToMany
+{
+    /**
+     * Get the key for comparing against the parent key in "has" query.
+     *
+     * @return string
+     */
+    public function getHasCompareKey()
+    {
+        return $this->getForeignKey();
+    }
+
+    /** @inheritdoc */
+    public function getRelationExistenceQuery(Builder $query, Builder $parentQuery, $columns = ['*'])
+    {
+        return $query;
+    }
+
+    /** @inheritdoc */
+    protected function hydratePivotRelation(array $models)
+    {
+        // Do nothing.
+    }
+
+    /**
+     * Set the select clause for the relation query.
+     *
+     * @return array
+     */
+    protected function getSelectColumns(array $columns = ['*'])
+    {
+        return $columns;
+    }
+
+    /** @inheritdoc */
+    protected function shouldSelect(array $columns = ['*'])
+    {
+        return $columns;
+    }
+
+    /** @inheritdoc */
+    public function addConstraints()
+    {
+        if (static::$constraints) {
+            $this->setWhere();
```

Side note: This make me realize that we don't support constraint on relational data. It could be possible using an aggregation `$lookup` stage. Should we throw an exception instead of hiding the fact that this library doesn't support it?

---

I'm not sure about that. If we throw an error in `addConstraints` method then we need another method to act like it. I mean this method will be called when we want to read the relation's data, not when we are working on the relation (like `sync`ing or any other actions). So, to create a method like `addConstraints`, we will need to override at least one method on the parent class. I think it requires too much effort to implement and update relationship classes while everything is working fine right now.